.github/workflows/test_asterinas.yml

```diff
@@ -19,21 +19,64 @@ jobs:
       - run: echo "Running in asterinas/asterinas:0.8.0"
 
       - uses: actions/checkout@v4
-      
+
       - name: Check
         id: check
         run: make check
 
-  unit-test:
+  build-initrd:
     if: github.event_name == 'push' || github.event_name == 'pull_request'
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: nixbuild/nix-quick-install-action@v28
+
+      - name: Restore and Cache Nix Store
+        uses: nix-community/cache-nix-action@v5
+        with:
+          primary-key: nix-${{ runner.os }}-${{ hashFiles('**/*.nix', 'flake.lock') }}
+          restore-prefixes-first-match: nix-${{ runner.os }}-
+          purge: true
+          purge-prefixes: nix-${{ runner.os }}-
+          purge-created: 0
+          purge-primary-key: never
+
+      - name: Build Initrd
+        run: |
+          nix build .#initrd -L
+
+      - name: Upload Initrd
+        uses: actions/upload-artifact@v4
+        with:
+          name: initrd.gz
+          path: result/initrd.gz
+          compression-level: 0  # initrd.gz is already compressed
+
+  unit-test:
+    needs: build-initrd
+    runs-on: ubuntu-latest
     timeout-minutes: 15
     container: asterinas/asterinas:0.8.0
+    env:
+      PREBUILT_INITRAMFS: result/initrd.gz
```

Comment:
> Why do we have to built initrd in a seperate job?

I ran into trouble when running Nix inside `container: asterinas/asterinas`, it's mostly because the user inside the container is `root`. Creating a non-root user in Dockerfile and running as it can solve this issue, but it seems a bit off topic.

> Is it because currently Nix will build too many things so the process to build initrd is slow?

No. Though the cold build does be slow (~0.5hr on Actions), but it's OK (~1min) with Actions cache.